drivers/serial/uart_mcux.c

```diff
@@ -315,7 +316,7 @@ static void uart_mcux_irq_callback_set(const struct device *dev,
 static void uart_mcux_isr(const struct device *dev)
 {
 	struct uart_mcux_data *data = dev->data;
-
+	pm_device_busy_clear(dev);
```

Comment:
are you sure about these busy_set()/busy_clear() calls ? They "look" unbalanced for me, you are setting busy at tx_enable() but not clearing it in tx_disabled(). But I'm not familiar with this device flow.

Reply:
I agree with @ceolin. Looks like the busy_clear should happen in tx_disable as it disables the TX interrupt which means your code to busy_clear will not happen. 
Also do you think the transmitter can be disabled when TX is disabled to save power?